shared/react-native/ios/Keybase/AppDelegate.m

```diff
@@ -229,15 +232,49 @@ - (void)applicationDidEnterBackground:(UIApplication *)application {
   [self.resignImageView.layer removeAllAnimations];
   // Snapshot happens right after this call, force alpha immediately w/o animation else you'll get a half animated overlay
   self.resignImageView.alpha = 1;
+  
+  const bool requestTime = KeybaseAppDidEnterBackground();
+  if (requestTime && (!self.shutdownTask || self.shutdownTask == UIBackgroundTaskInvalid)) {
+    UIApplication *app = [UIApplication sharedApplication];
+    self.shutdownTask = [app beginBackgroundTaskWithExpirationHandler:^{
+      KeybaseAppWillExit();
+      [app endBackgroundTask:self.shutdownTask];
+      self.shutdownTask = UIBackgroundTaskInvalid;
+    }];
+  }
 }
 
 // Sometimes these lifecycle calls can be skipped so try and catch them all
 - (void)applicationDidBecomeActive:(UIApplication *)application {
   [self hideCover];
+  [self notifyAppState:application];
 }
 
 - (void)applicationWillEnterForeground:(UIApplication *)application {
   [self hideCover];
+  KeybaseSetAppStateBackground();
+}
+
+- (void)applicationDidFinishLaunching:(UIApplication *)application {
+  [self notifyAppState:application];
+}
+
+- (void)notifyAppState:(UIApplication *)application {
+  const UIApplicationState state = application.applicationState;
+  switch (state) {
+    case UIApplicationStateActive:
+      KeybaseSetAppStateForeground();
+      break;
+    case UIApplicationStateBackground:
+      KeybaseSetAppStateBackground();
+      break;
+    case UIApplicationStateInactive:
+      KeybaseSetAppStateInactive();
+      break;
+    default:
+      KeybaseSetAppStateForeground();
```

Comment:
this is to match the behavior on [this line](https://github.com/keybase/client/pull/11142/files#diff-25ab83d9513d1ae61c1f3a4d40b0f15fL22)